_drafts/STMViewer.md

```diff
@@ -0,0 +1,167 @@
+
+## What exactly can it visualize?
+
+STMViewer consists of two modules which serve a slightly different purpose: 
+
+### Variable Viewer
+Variable Viewer is a module that reads variables' values directly from RAM using ST-Link SWD (only SWDIO, SWCLK, and GND connection is needed). It is asynchronous, which means it samples the memory with the time period set on the PC. It is completely non-intrusive and can log many variables at once. Writing is supported as well. There are some trade-offs though. The first one is that logged variables' addresses have to be constant, meaning they have to be globals. Moreover, with more aggressive optimization some of the variables may get optimized and fail to be logged.
+
+### Trace Viewer 
+Trace Viewer is a module that reads and parses SWO (Serial Wire Viewer) data using an ST-Link. To use it, you need to connect SWDIO, SWCLK, SWO, and GND. This tool operates synchronously, meaning it visualizes data points at the rate they are produced on the target. A data point can be either a special single-byte flag used to create digital plots or any value that fits within up to 4 bytes, as in the case of 'analog' plots. It incurs minimal performance overhead, involving just a single register write per plot point, as serialization and timestamping are handled by the ITM peripheral. Trace Viewer can be employed for profiling specific parts of the codebase or visualizing signals that are too fast for the asynchronous Variable Viewer. Its main limitations are the maximum allowed SWO pin baudrate of the ST-Link and the fact that the ITM peripheral is supported only on Cortex M3/M4/M7/M33 cores. 
+
+
+## I'm already using STMStudio/CubeMonitor, why create a new tool?
+
+Let's begin with the most commonly asked question: Why bother creating a tool when existing options are provided by ST? There are several reasons:
+
+STMStudio is deprecated, and while it was a useful tool, it became challenging to work with due to persistent bugs. It also lacks support for mangled C++ names and is exclusively available for Windows.
+
+CubeMonitor, on the other hand, is often considered too much overhead for debugging purposes. It's primarily designed for creating dashboards, and during debugging, the focus tends to be on functionality rather than aesthetics. Manipulating plot data and writing values to logged variables can be inconvenient. Furthermore, setting up a basic logger configuration can require some time to become familiar with NodeRED's setup.
+
+Last but not least, none of the available hobby-level software tools include a trace data visualizer. In my opinion, STMViewer's trace data visualizer is the most intriguing module, enabling users to visualize variables synchronously and profile functions or interrupts within minutes.
+
+## Variable Viewer
+
+<!-- TODO GIF ? -->
```

Comment:
Hi Piotr! Did you intend to add gif or images in these `TODO`s? If not, could you remove these `TODO` comments?

Reply:
Hello! You're right, I forgot to remove these comments. It should be ok now. I added the GIF's, but I'm not sure if they aren't too much, when we consider the text-to-images ratio. Please let me know if I should keep or remove them. Is there an option to make the images clickable to zoom them? I get the impression that the position_PID image I've changed is still not readable, and the same goes for the GIFs. They are fine when zoomed in, so it's not a matter of poor resolution.